tests/generation/test_generation_utils.py

```diff
@@ -1693,6 +1693,25 @@ def test_diverse_beam_search(self):
             ],
         )
 
+    @slow
+    def test_contrastive_search(self):
```

Comment:
If possible I'd be really happy if we could also test this on BART, T5 and GPT-J . Then we should have covered 95% of the model architectures. But ok to do in a follow-up PR . Currently I don't expect the method to work with T5.